packages/pros/src/devices/controller.rs

```diff
@@ -3,333 +3,260 @@
 //! Controllers are identified by their id, which is either 0 (master) or 1 (partner).
 //! State of a controller can be checked by calling [`Controller::state`] which will return a struct with all of the buttons' and joysticks' state.
 
-use alloc::{ffi::CString, vec::Vec};
+use alloc::ffi::CString;
 
-use pros_sys::{controller_id_e_t, PROS_ERR};
+use pros_sys::{E_CONTROLLER_MASTER, E_CONTROLLER_PARTNER, PROS_ERR};
 use snafu::Snafu;
 
 use crate::error::{bail_on, map_errno};
 
+pub const CONTROLLER_MAX_LINE_LENGTH: usize = 14;
```

Comment:
Maybe this should go in pros-sys because it determines constraints on a function from that module. Also if controller display gets implemented in sim it'd be good to have that constant available because the simulator already uses other constants from pros-sys like errno values.